core/src/main/java/org/mskcc/cbio/portal/dao/DaoReferenceGenomeGene.java

```diff
@@ -0,0 +1,255 @@
+            pstmt.setInt(2, referenceGenomeId);
+            rs = pstmt.executeQuery();
+            if (rs.next()) {
+                return extractGene(rs);
+            } else {
+                return null;
+            }
+        } catch (SQLException e) {
+            throw new DaoException(e);
+        } finally {
+            JdbcUtil.closeAll(DaoGene.class, con, pstmt, rs);
+        }
+    }
+
+    /**
+     * Gets the Gene with the Specified Entrez Gene ID.
+     * For faster access, consider using DaoGeneOptimized.
+     *
+     * @param hugoGeneSymbol Hugo Gene Symbol.
+     * @return Canonical Gene Object.
+     * @throws DaoException Database Error.
+     */
+    public ReferenceGenomeGene getGene(String hugoGeneSymbol, int referenceGenomeId) throws DaoException {
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoReferenceGenomeGene.class);
+            pstmt = con.prepareStatement
+                ("SELECT * FROM `reference_genome_gene` JOIN `gene` ON `reference_genome_gene`.entrez_gene_id=" +
+                    "`gene`.entrez_gene_id WHERE `HUGO_GENE_SYMBOL` = ? AND `REFERENCE_GENOME_ID` = ?");
+            pstmt.setString(1, hugoGeneSymbol);
+            pstmt.setInt(2, referenceGenomeId);
+            rs = pstmt.executeQuery();
+            if (rs.next()) {
+                return extractGene(rs);
+            } else {
+                return null;
+            }
+        } catch (SQLException e) {
+            throw new DaoException(e);
+        } finally {
+            JdbcUtil.closeAll(DaoGene.class, con, pstmt, rs);
+        }
+    }
+
+    private ReferenceGenomeGene extractGene(ResultSet rs) throws SQLException, DaoException {
+        int entrezGeneId = rs.getInt("ENTREZ_GENE_ID");
+        int reference_genome_id = rs.getInt("REFERENCE_GENOME_ID");
+        String cytoband = rs.getString("CYTOBAND");
+        ReferenceGenomeGene gene = new ReferenceGenomeGene(entrezGeneId, reference_genome_id);
+        gene.setChr(rs.getString("CHR"));
+        gene.setCytoband(rs.getString("CYTOBAND"));
+        gene.setExonicLength(rs.getInt("EXONIC_LENGTH"));
+        gene.setStart(rs.getLong("START"));
+        gene.setEnd(rs.getLong("END"));
+        return gene;
+    }
+
+    /**
+     * Deletes the Reference Genome Gene Record with Entrez Gene ID and Referece Genome ID in the Database.
+     *
+     * @param entrezGeneId ENTREZ GENE ID 
+     * @param referenceGenomeId REFERENCE GENOME ID
+     */
+    public void deleteGene(int entrezGeneId, int referenceGenomeId) throws DaoException {
+
+        Connection con = null;
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        try {
+            con = JdbcUtil.getDbConnection(DaoReferenceGenomeGene.class);
+            pstmt = con.prepareStatement("DELETE FROM `reference_genome_gene` WHERE ENTREZ_GENE_ID=? AND REFERENCE_GENOME_ID=?");
```

Comment:
we probably need functionality to be added to DaoGene ... which will delete all linked ReferenceGenomeGene records for an entrez_gene_id when you delete that entrez_gene_id from the gene table.

Reply:
we do not need one here since I had a **_cascade_** deletion foreign key i.e. when a gene deleted all reference genome genes will be automatically deleted as well.